File: src/main/java/iuh/fit/se/phazelsound/modules/auth/service/AuthService.java
package iuh.fit.se.phazelsound.modules.auth.service;

import iuh.fit.se.phazelsound.modules.auth.dto.request.LoginUserRequest;
import iuh.fit.se.phazelsound.modules.auth.dto.request.RegisterUserRequest;
import iuh.fit.se.phazelsound.modules.auth.dto.request.ResetPasswordRequest;
import iuh.fit.se.phazelsound.modules.auth.dto.response.AuthResponse;

public interface AuthService {
    String register(RegisterUserRequest request);
    String verifyRegisterOtp(String email, String otp);
    AuthResponse login(LoginUserRequest request);

    String resendRegisterOtp(String email);
    String sendForgotPasswordOtp(String email);
    String resetPassword(ResetPasswordRequest request);
}

File: PhazelSoundMobile/src/features/auth/auth.service.ts
import axiosClient from '../../api/axiosClient';
import {
  RegisterRequest,
  LoginRequest,
  AuthResponse,
  ResetPasswordRequest,
} from './auth.types';

class AuthService {
  async register(data: RegisterRequest): Promise<string> {
    const response = await axiosClient.post('/auth/register', data);
    return response.data;
  }

  async verifyOtp(email: string, otp: string): Promise<string> {
    const response = await axiosClient.post('/auth/verify', null, {
      params: { email, otp },
    });
    return response.data;
  }

  async login(data: LoginRequest): Promise<AuthResponse> {
    const response = await axiosClient.post<AuthResponse>('/auth/login', data);
    return response.data;
  }

  async resendRegisterOtp(email: string): Promise<string> {
    const response = await axiosClient.post('/auth/resend-register-otp', null, {
      params: { email },
    });
    return response.data;
  }

  async forgotPassword(email: string): Promise<string> {
    const response = await axiosClient.post('/auth/forgot-password', null, {
      params: { email },
    });
    return response.data;
  }

  async resetPassword(data: ResetPasswordRequest): Promise<string> {
    const response = await axiosClient.post('/auth/reset-password', data);
    return response.data;
  }
}

export default new AuthService();
